src/NumericalAlgorithms/Spectral/SwshTransformJob.hpp

```diff
@@ -0,0 +1,380 @@
+      for (size_t i = 0; i < number_of_radial_grid_points_; ++i) {
+        post_transform_slices.push_back(detail::ComplexDataView<Representation>{
+            make_not_null(&output_vector), collocation_metadata_->size(),
+            i * collocation_metadata_->size()});
+        post_transform_collocation_data.push_back(
+            post_transform_slices.back().real_data());
+        post_transform_collocation_data.push_back(
+            post_transform_slices.back().imag_data());
+        // libsharp considers two arrays per transform when spin is not zero.
+        num_transforms += (spin == 0 ? 2 : 1);
+      }
+    });
+
+    sharp_execute(SHARP_ALM2MAP, abs(spin),
+                  pre_transform_coefficient_data.data(),
+                  post_transform_collocation_data.data(),
+                  collocation_metadata_->get_sharp_geom_info(), alm_info_,
+                  num_transforms, SHARP_DP, nullptr, nullptr);
+
+    if (spin < 0) {
+      // the conjugate is needed because libsharp doesn't handle negative spins.
+      for (size_t i = 0; i < post_transform_slices.size(); ++i) {
+        post_transform_slices[i].conjugate();
+      }
+    }
+
+    // The inverse transformed collocation data has just been placed in the
+    // memory blocks controlled by the `ComplexDataView`s. Finally, that data
+    // must be flushed back to the Variables.
+    for (size_t i = 0; i < post_transform_slices.size(); ++i) {
+      post_transform_slices[i].copy_back_to_source();
+    }
+  }
+
+ private:
+  size_t number_of_radial_grid_points_;
+  size_t l_max_;
+  sharp_alm_info* alm_info_;
+  const Collocation<Representation>* collocation_metadata_;
+};
+
+namespace detail {
+template <typename Job>
+struct transform_job_is_not_empty : std::true_type {};
+
+template <int Spin, ComplexRepresentation Representation>
+struct transform_job_is_not_empty<
+    TransformJob<Spin, Representation, tmpl::list<>>> : std::false_type {};
+
+template <int MinSpin, ComplexRepresentation Representation, typename TagList,
+          typename IndexSequence>
+struct make_swsh_transform_job_list_impl;
+
+template <int MinSpin, ComplexRepresentation Representation, typename TagList,
+          int... Is>
+struct make_swsh_transform_job_list_impl<MinSpin, Representation, TagList,
+                                         std::integer_sequence<int, Is...>> {
+  using type = tmpl::filter<
+      tmpl::list<TransformJob<Is + MinSpin, Representation,
+                              get_tags_with_spin<Is + MinSpin, TagList>>...>,
+      transform_job_is_not_empty<tmpl::_1>>;
+};
+}  // namespace detail
+
+/// \ingroup SpectralGroup
+/// \brief Assemble a `tmpl::list` of `TransformJob` given a list of tags
+/// `TagList that need to be transformed. The `Representation` is the
+/// `ComplexRepresentation` to use for the transformations.
+///
+/// \details Up to five `TransformJob` will be returned, corresponding to
```

Comment:
Presumably it is just because I don't know enough of SWSHs, but why 5?

Reply:
explicitly limited to spin-weights range -2 to 2. It's a library limitation, but one that's fine for current application